src/lib/bmp/bmp.c

```diff
@@ -0,0 +1,360 @@
+					image[pos + 1] = pallete[buff[0]].green;
+					image[pos + 2] = pallete[buff[0]].red;
+					pos += 3;
+				}
+				if (skip) {
+					read(fd, buff, skip);
+				}
+			}
+			break;
+
+		case 4:
+			read_pallete(fd, pallete, (bmp_head->bfOffBits - 54) / 4);
+			lseek(fd, bmp_head->bfOffBits, SEEK_SET);
+			skip = FIND_SKIP(width / 2 + width % 2);
+			for (i = 0; i < height; i++) {
+				for (j=0; j < width / 2; j++) {
+					read(fd, buff, 1);
+					buff[1] = buff[0]>>4;
+					buff[2] = buff[0] & 0x0f;
+					image[pos + 5] = pallete[buff[1]].red;
+					image[pos + 4] = pallete[buff[1]].green;
+					image[pos + 3] = pallete[buff[1]].blue;
+					image[pos + 2] = pallete[buff[2]].red;
+					image[pos + 1] = pallete[buff[2]].green;
+					image[pos] = pallete[buff[2]].blue;
+					pos += 6;
+				}
+				if (width % 2) {
+					read(fd, buff, 1);
+					buff[1] = buff[0]>>4;
+					image[pos + 2] = pallete[buff[1]].red;
+					image[pos + 1] = pallete[buff[1]].green;
+					image[pos] = pallete[buff[1]].blue;
+					pos += 3;
+				}
+				if (skip) {
+					read(fd, buff, skip);
+				}
+			}
+			break;
+
+		case 1:
+			lseek(fd, bmp_head->bfOffBits, SEEK_SET);
+			skip = FIND_SKIP(width / 8 + (width % 8 ? 1 : 0));
+			for (i = 0; i < height; i++) {
+				for (j = 0; j < width/8; j++) {
+					read(fd, buff, 1);
+					for (k =0; k < 8; k++) {
+						if (buff[0] & 0x80) {
+							image[pos] = 255;
+							image[pos + 1] = 255;
+							image[pos + 2] = 255;
+							image[pos + 3] = 255;
+							pos += 3;
+						} else {
+							image[pos] = 0;
+							image[pos + 1] = 0;
+							image[pos + 2] = 0;
+							pos += 3;
+						}
+						buff[0] = buff[0]<<1;
+					}
+				}
+
+				if (width % 8) {
+					read(fd, buff, 1);
+					for (k = 0; k < width % 8; k++) {
+						if (buff[0] & 0x80) {
+							image[pos] = 255;
+							image[pos + 1] = 255;
+							image[pos + 2] = 255;
+							pos += 3;
+						} else {
+							image[pos] = 0;
+							image[pos + 1] = 0;
+							image[pos + 2] = 0;
+							pos += 3;
+						}
+						buff[0] = buff[0]<<1;
+					}
+				}
+
+				if (skip) {
+					read(fd, buff, skip);
+				}
+			}
+			break;
+
+		default:
+			printf("invalid bitcount\n");
+			return -1;
+	}
+	return 0;
+}
+
+static int get_fbp_bmp(struct fb_var_screeninfo *vinfo, char *buf, uint8_t *img,
+									uint32_t width, uint32_t height, uint32_t bits) {
+	int i, j, offset, addres, x = 0;
+	int fb_bytes_per_pixel = vinfo->bits_per_pixel / 8;
+
+	for (i = height; i > 0; i--) {
+		offset = 0;
+		for (j = width; j > 0; j--) {
+			addres = offset + i * vinfo->xres * fb_bytes_per_pixel;
+			offset += fb_bytes_per_pixel;
+
+			if (vinfo->bits_per_pixel == 32) {
+				* (buf + addres) = img[x];
+				* (buf + addres + 1) = img[x + 1];
+				* (buf + addres + 2) = img[x + 2];
+				* (buf + addres + 3) = 255;
+				x += 3;
+			} else {
+				printf("Error : it's draw only in 32bpp framebuffer\n");
+				return -1;
+			}
+		}
+	}
+	return 0;
+}
+
+int draw_fb_bmp(char *file_name) {
```

Comment:
This function looks very specific, I guess it's better to make it a separate command.

Reply:
okey, I will remove this function from the library and create a new PR,
can you give advice on which section to add this fun

Reply:
deleted